src/searcher_frame_document.rs

```diff
@@ -56,28 +31,41 @@ impl SearchResult {
     }
 
     #[getter]
-    fn unique_docs(&self, py: Python) -> PyResult<Vec<u64>> {
+    fn unique_docs(&self, py: Python) -> PyResult<BTreeSet<u64>> {
         let s =
             BTreeSet::from_iter(self.hits.iter().map(|(d, f, s, score)| *d));
-        let v = Vec::from_iter(s.into_iter());
-        Ok(v)
+        Ok(s)
     }
 
     #[getter]
-    fn unique_frames(&self, py: Python) -> PyResult<Vec<u64>> {
+    fn unique_frames(&self, py: Python) -> PyResult<BTreeSet<u64>> {
         let s =
             BTreeSet::from_iter(self.hits.iter().map(|(d, f, s, score)| *f));
-        let v = Vec::from_iter(s.into_iter());
-        Ok(v)
+        Ok(s)
     }
 
     #[getter]
-    fn unique_docs_frames(&self, py: Python) -> PyResult<Vec<(u64, u64)>> {
+    fn unique_docs_frames(&self, py: Python) -> PyResult<BTreeSet<(u64, u64)>> {
         let s = BTreeSet::from_iter(
             self.hits.iter().map(|(d, f, s, score)| (*d, *f)),
         );
-        let v = Vec::from_iter(s.into_iter());
-        Ok(v)
+        Ok(s)
+    }
+
+    /// This is an optimization to allow Python callers to obtain vectors
+    /// without having to do iteration to get them.
+    #[getter]
+    fn unique_docs_frames_unzipped(&self, py: Python) -> PyResult<(Vec<u64>, Vec<u64>)> {
```

Comment:
This is an optimization idea I had. Basically, by returning a vec of matched docs and a vec of matched frames, it means that Python code would not need iteration to extract this data out of the set of tuples as returned by `unique_docs_frames`.  The idea is less iteration in Python as a performance improvement.

However, for datasets that have multiple frames per doc, the implementation I have here would contain duplicates in the "docs" vector.  I think that's ok, because we do want the two vecs to actually "match up" I think.  The "unique" in the name of the function still refers to the pairs, not the contents of each returned vec.

I could change this function to include a 3rd vec for sentences...what do you think?